### PR TITLE
feat: containerd config change for artifact streaming

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This changes the containerd config to use overlaybd if artifact streaming is enabled. 
It also adds multiple tests to check the config. 

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -1166,6 +1166,10 @@ root = "{{GetDataDir}}"{{- end}}
     {{- if TeleportEnabled }}
     snapshotter = "teleportd"
     disable_snapshot_annotations = false
+    {{- end}}
+	{{- if IsArtifactStreamingEnabled }}
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:1170]: extra indent here?

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -1256,6 +1266,10 @@ root = "{{GetDataDir}}"{{- end}}
...
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:1270]: also extra indent here?

File: pkg/agent/baker_test.go [hunk 3]
```diff
@@ -916,6 +916,79 @@ oom_score = 0
...
+`
+				Expect(containerdConfigFileContent).To(ContainSubstring(expectedOverlaybdConfig))
+				expectedOverlaybdPlugin := `[proxy_plugins]
+  [proxy_plugins.overlaybd]
+    type = "snapshot"
+    address = "/run/overlaybd-snapshotter/overlaybd.sock"`
+				Expect(containerdConfigFileContent).To(ContainSubstring(expectedOverlaybdPlugin))
+			},
+		),
+		Entry("AKSUbuntu2204 without artifact streaming", "AKSUbuntu1804+NoArtifactStreaming", "1.25.7", func(config *datamodel.NodeBootstrappingConfiguration) {
```
Comment (cameronmeissner) [inline pkg/agent/baker_test.go:968]: linty :(

**Review by cameronmeissner — APPROVED** [review]

File: pkg/agent/baker.go (hunk 1 above)
Comment (ganeshkumar5699) [reply pkg/agent/baker.go]: yes, tabs vs spaces haha, so it looked the same on VS code

File: pkg/agent/baker_test.go (hunk 3 above)
Comment (ganeshkumar5699) [reply pkg/agent/baker_test.go]: reduced test description characters for linting

File: pkg/agent/baker_test.go [hunk 4]
```diff
@@ -916,6 +916,79 @@ oom_score = 0
...
+`
+				Expect(containerdConfigFileContent).To(ContainSubstring(expectedOverlaybdConfig))
+				expectedOverlaybdPlugin := `[proxy_plugins]
+  [proxy_plugins.overlaybd]
+    type = "snapshot"
+    address = "/run/overlaybd-snapshotter/overlaybd.sock"`
+				Expect(containerdConfigFileContent).To(ContainSubstring(expectedOverlaybdPlugin))
+			},
+		),
+		Entry("AKSUbuntu2204 w/o artifact streaming", "AKSUbuntu1804+NoArtifactStreaming", "1.25.7", func(config *datamodel.NodeBootstrappingConfiguration) {
```
Comment (cameronmeissner) [inline pkg/agent/baker_test.go:968]: linty is having a very bad day

**Review by cameronmeissner — APPROVED** [review]

File: pkg/agent/baker_test.go (hunk 4 above)
Comment (ganeshkumar5699) [reply pkg/agent/baker_test.go:968]: changed linter config to have 180 character limit

**Pull request merged**